### PR TITLE
Setup Sentry for API

### PR DESCRIPTION
**Pull request by arku**
# 💅🏼 What issue does this fix?

Capture exceptions in the API using [sentry-raven](https://github.com/getsentry/raven-ruby).

## 🍂 Before You Submit

* [ ] Did you write tests?
* [ ] Did you run `rails rswag` from the root?
* [x] Did you run `rubocop` from the root?
* [x] Did you run `yarn lint` in `/client`?
* [x] Did you run `yarn test` in `/client`?
* [ ] Are your primary keys UUIDs on any new tables?

## 🛷 Deployment Considerations

* [ ] Data Migrations
* [ ] Schema Migrations
* [x] Dependencies

To do:
- Create a Rails project in our Sentry account
- Add `REACT_APP_SENTRY_DSN_FRONTEND` and `SENTRY_DSN_BACKEND` environment variables.

## 🧵 Steps to set up locally

Run `bundle`

## 🕯 Caveats, concerns

Sentry is only enabled on Production at the moment. You'll have to make some code changes to test if the exceptions are sent to Sentry.

### CODE REVIEW

File: config/initializers/sentry.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+if Rails.env.production?
+  Raven.configure do |config|
+    config.dsn = ENV.fetch('SENTRY_DSN', '')
```
Comment (katelovescode) [inline config/initializers/sentry.rb:5]: We're going to need a different env var; we currently have `SENTRY_DSN` for the frontend project.  Can we change both of them to `SENTRY_DSN_FRONTEND` & `SENTRY_DSN_BACKEND`?

Comment (arku) [reply config/initializers/sentry.rb]: We have `REACT_APP_SENTRY_DSN` on the frontend but I made the change anyway.

Comment (katelovescode) [reply config/initializers/sentry.rb]: Hm.  My ENV VARS might have gotten out of date.

File: client/.env.sample [hunk 2]
```diff
@@ -1 +1 @@
-REACT_APP_SENTRY_DSN=[leave empty on development]
+REACT_APP_SENTRY_DSN_FRONTEND=[leave empty on development]
```
Comment (katelovescode) [inline client/.env.sample:1]:
```suggestion
REACT_APP_SENTRY_DSN=[leave empty on development]
```
Ignore me, I forgot these are kept in different .env files 🙄  it's been a long week for me, @arku don't judge me 😂

File: .env.sample [hunk 3]
```diff
@@ -10,7 +10,7 @@ PGPASSWORD=yourlocalpostgrespassword (optional, in case your local postgres inst
 MAIL_URL= # localhost:3000 (or other mail provider url)
 
 # only for staging and production environments
-SENTRY_DSN= # staging and later envs
+SENTRY_DSN_BACKEND= # staging and later envs
```
Comment (katelovescode) [inline .env.sample:13]:
```suggestion
RAILS_APP_SENTRY_DSN= # staging and later envs
```

File: client/src/index.js [hunk 4]
```diff
@@ -6,9 +6,12 @@ import './tailwind.generated.css'
 import * as Sentry from '@sentry/browser'
 import './i18n'
 
-if (process.env.NODE_ENV === 'production' && process.env.REACT_APP_SENTRY_DSN) {
+if (
+  process.env.NODE_ENV === 'production' &&
+  process.env.REACT_APP_SENTRY_DSN_FRONTEND
```
Comment (katelovescode) [inline client/src/index.js:11]:
```suggestion
  process.env.REACT_APP_SENTRY_DSN
```

File: client/src/index.js [hunk 5]
```diff
@@ -6,9 +6,12 @@ import './tailwind.generated.css'
...
 import './i18n'
 
-if (process.env.NODE_ENV === 'production' && process.env.REACT_APP_SENTRY_DSN) {
+if (
+  process.env.NODE_ENV === 'production' &&
+  process.env.REACT_APP_SENTRY_DSN
+) {
   Sentry.init({
-    dsn: process.env.REACT_APP_SENTRY_DSN
+    dsn: process.env.REACT_APP_SENTRY_DSN_FRONTEND
```
Comment (katelovescode) [inline client/src/index.js:14]:
```suggestion
    dsn: process.env.REACT_APP_SENTRY_DSN
```

File: config/initializers/sentry.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+if Rails.env.production?
+  Raven.configure do |config|
+    config.dsn = ENV.fetch('SENTRY_DSN_BACKEND', '')
```
Comment (katelovescode) [inline config/initializers/sentry.rb:5]:
```suggestion
    config.dsn = ENV.fetch('RAILS_APP_SENTRY_DSN', '')
```

**Pull request merged**